File: TUF/Array/FindAllRptElm.java
import java.util.HashMap;
import java.util.Scanner;
import java.util.Map.Entry;

public class FindAllRptElm {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the size of the array: ");
        int size = input.nextInt();
        int[] arr = new int[size];
        for (int i = 0; i < size; i++) {
            System.out.printf("Enter %d number element: ", i + 1);
            arr[i] = input.nextInt();
        }
        input.close();
        System.out.println("The repeating elements are: ");
        findRepeatingElements(arr);
    }

    private static void findRepeatingElements(int[] arr) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < arr.length; i++) {
            map.put(arr[i], map.getOrDefault(arr[i], 0) + 1);
        }
        for (Entry<Integer, Integer> entry : map.entrySet()) {
            if (entry.getValue() > 1) {
                System.out.print(entry.getKey() + ", ");
            }
        }
    }
}

File: TUF/Array/FindAllRptElm.ts
function findRepeatingElements(arr: number[]): void {
    let map = new Map<number, number>();
    for (let i: number = 0; i < arr.length; i++) {
        map.set(arr[i], (map.get(arr[i]) || 0) + 1);
    }
    let resultArray: number[] = [];
    for (let [key, value] of map.entries()) {
        if (value > 1) {
            resultArray.push(key);
        }
    }
    console.log("The repeating elements are: ", resultArray);
}

let arr: number[] = [1, 2, 4, 3, 3, 4, 5, 6, 7, 8, 9];
findRepeatingElements(arr);

export { };